applications/GeoMechanicsApplication/custom_constitutive/truss_backbone_constitutive_law.cpp

```diff
@@ -0,0 +1,166 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//
+//  Main authors:    Wijtze Pieter Kikstra
+//
+
+// Project includes
+#include "truss_backbone_constitutive_law.h"
+#include "geo_mechanics_application_variables.h"
+#include "includes/properties.h"
+
+namespace Kratos
+{
+ConstitutiveLaw::Pointer TrussBackboneConstitutiveLaw::Clone() const
+{
+    return Kratos::make_shared<TrussBackboneConstitutiveLaw>(*this);
+}
+
+void TrussBackboneConstitutiveLaw::GetLawFeatures(Features& rFeatures)
+{
+    // Set the strain size
+    rFeatures.mStrainSize = 1;
+
+    // Set the space dimension
+    rFeatures.mSpaceDimension = 3;
+}
+
+double& TrussBackboneConstitutiveLaw::GetValue(const Variable<double>& rThisVariable, double& rValue)
+{
+    if (rThisVariable == ACCUMULATED_STRAIN) {
+        rValue = mAccumulatedStrain;
+    } else {
+        KRATOS_ERROR << "Can't get the specified value for " << rThisVariable << std::endl;
+    }
+    return rValue;
+}
+
+array_1d<double, 3>& TrussBackboneConstitutiveLaw::GetValue(const Variable<array_1d<double, 3>>& rThisVariable,
+                                                            array_1d<double, 3>& rValue)
+{
+    KRATOS_ERROR << "Can't get the specified value for " << rThisVariable << std::endl;
+}
```

Comment:
IMO if you do not have anything to return, do not override it and even worse return an error. Imagine that you have 2 materials (this CL and another one in which printing the double var is relevant) in one simulation, if you try to print posts of a `double` type variable your code will crash because of this.

Reply:
Let me explain the reasoning at least within the SMApp:

- Each time you want to print anything at IP level, the` element.CalculatoOnIntegrationPoints()` is called.
- Then, this element method checks wether the `CL.Has()`  this variable (this method has to be implemented in the CL)
- If the `CL.Has()` is true then return the `CL.GetValue()`; if not return `CL.CalculateValue()`.
- If the CL is not related or does not know how to calculate something, it just does nothing, do not throw an error because of hat I said before.

Reply:
IDEM with `element.SetValuesOnIntegrationPoints()`